lib/web_ui/lib/src/engine/semantics/semantics_helper.dart

```diff
@@ -0,0 +1,316 @@
+  /// that's being removed. Chrome doesn't have this issue.
+  bool _schedulePlaceholderRemoval = false;
+
+  /// Whether we are waiting for the user to enable semantics.
+  bool get _isWaitingToEnableSemantics => _semanticsPlaceholder != null;
+
+  bool shouldEnableSemantics(html.Event event) {
+    if (!_isWaitingToEnableSemantics) {
+      // Forward to framework as normal.
+      return true;
+    } else {
+      return tryEnableSemantics(event);
+    }
+  }
+
+  /// Attempts to activate semantics.
+  ///
+  /// Returns true if the `event` is not related to semantics activation and
+  /// should be forwarded to the framework.
+  bool tryEnableSemantics(html.Event event);
+
+  /// Creates the placeholder for accesibility.
+  ///
+  /// Puts it inside the glasspane.
+  ///
+  /// On focus the element announces that accessibility can be enabled by
+  /// tapping/clicking. (Announcement depends on the assistive technology)
+  html.Element prepareAccesibilityPlaceholder();
+}
+
+@visibleForTesting
+class DesktopEnableSemantics extends _EnableSemantics {
+  @override
+  bool tryEnableSemantics(html.Event event) {
+    if (_schedulePlaceholderRemoval) {
+      _semanticsPlaceholder.remove();
+      _semanticsPlaceholder = null;
+      semanticsActivationTimer = null;
+      return true;
+    }
+
+    if (EngineSemanticsOwner.instance.semanticsEnabled) {
+      // Semantics already enabled, forward to framework as normal.
+      return true;
+    }
+
+    // In touch screen laptops, the touch is received as a mouse click
+    const List<String> kInterestingEventTypes = <String>[
+      'click',
+      'keyup',
+      'keydown',
+      'mouseup',
+      'mousedown',
+      'pointerdown',
+      'pointerup',
+    ];
+
+    if (!kInterestingEventTypes.contains(event.type)) {
+      // The event is not relevant, forward to framework as normal.
+      return true;
+    }
+
+    semanticsActivationAttempts += 1;
+    if (semanticsActivationAttempts >= kMaxSemanticsActivationAttempts) {
+      // We have received multiple user events, none of which resulted in
+      // semantics activation. This is a signal that the user is not interested
+      // in semantics, and so we will stop waiting for it.
+      _schedulePlaceholderRemoval = true;
+      return true;
+    }
+
+    if (semanticsActivationTimer != null) {
+      // We are in a waiting period to activate a timer. While the timer is
+      // active we should consume events pertaining to semantics activation.
+      // Otherwise the event will also be interpreted by the framework and
+      // potentially result in activating a gesture in the app.
+      return false;
+    }
+
+    // Check for the event target.
+    bool enableConditionPassed = (event.target == _semanticsPlaceholder);
+
+    if (enableConditionPassed) {
+      assert(semanticsActivationTimer == null);
+      semanticsActivationTimer = Timer(const Duration(milliseconds: 300), () {
```

Comment:
What is the 300 ms delay for?

Reply:
300 value is random from the previous implementation.

It is a timer for consuming all the events in the engine and stop sending them to framework.

Reply:
Thanks! I carried to a common const value.